src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(operand) {
+      calculator.result += operand;
+
+      return calculator;
+    },
+    subtract(operand) {
+      calculator.result -= operand;
+
+      return calculator;
```

Comment:
do not return calculator in the methods except `operate` and `reset`
you are returning calculator in the end, and that's may be the reason for failing tests

Reply:
function makeCalculator() {
  const calculator = {
    result: 0,
    add(operand) {
      this.result += operand;

      return this;
    },
    subtract(operand) {
      this.result -= operand;

      return this;
    },
    multiply(operand) {
      this.result *= operand;

      return this;
    },
    divide(operand) {
      this.result /= operand;

      return this;
    },
    reset() {
      this.result = 0;

      return this;
    },
    operate(callback, operand) {
      callback.call(this, operand);

      return this;
    },
  };

  return calculator;
}

Reply:
This is the solution Oleksandr helped me to achieve and it is working.
Do you want to change this working solution to next one?

Reply:
function makeCalculator() {
  const calculator = {
    result: 0,
    add(operand) {
      calculator.result += operand;
    },
    subtract(operand) {
      calculator.result -= operand;
    },
    multiply(operand) {
      calculator.result *= operand;
    },
    divide(operand) {
      calculator.result /= operand;
    },
    reset() {
      this.result = 0;

      return this;
    },
    operate(callback, operand) {
      callback.call(this, operand);

      return this;
    },
  };

  return calculator;
}